src/modules/thirdParty/sagas.ts

```diff
@@ -110,4 +160,56 @@ export function* thirdPartySaga(builder: BuilderAPI) {
   function* handleBuyThirdPartyItemSlotSuccess() {
     yield put(closeModal('BuyItemSlotsModal'))
   }
+
+  function* handlePublishThirdPartyItemRequest(action: PublishThirdPartyItemsRequestAction) {
+    const { thirdParty, items } = action.payload
+    try {
+      const collectionId = items[0].collectionId!
+
+      const { signature, signedMessage, salt } = yield call(getPublishItemsSignature, thirdParty.id, items.length)
+
+      const { items: newItems, itemCurations: newItemCurations }: { items: Item[]; itemCurations: ItemCuration[] } = yield call(
+        [builder, 'publishCollection'],
+        collectionId,
+        items.map(i => i.id),
+        signedMessage,
+        signature,
+        items.length, // qty
+        salt
```

Comment:
Not for now, but I think [this comment](https://github.com/decentraland/builder-server/pull/419#discussion_r822101180) applies for this line too!

Reply:
This remains the same here, but it's all wrapped in the `cheque` object in the `publishTPCollection` method.